tests/test_session.py

```diff
@@ -1,10 +1,34 @@
+import os
 import tempfile
 import unittest
+from contextlib import contextmanager
 from pathlib import Path
+from typing import Generator
```

Comment:
```suggestion
from typing import Iterator
```

Reply:
lineno 2-29 @ `tests/test_session.py`
```python3
import tempfile
from contextlib import contextmanager
from pathlib import Path
from typing import Generator


@contextmanager  # <~~~ context managers always yields a generator.
def reopenable_temp_file(content: str) -> Generator:
    """Reopenable tempfile to support writing/reading to/from the opened
    tempfile (requiered for Windows OS).

    For more information: https://bit.ly/3cr0Qkl

    :param content: string content to write.
    :yields: tempfile path.
    """
    try:
        with tempfile.NamedTemporaryFile(
            mode="w", suffix=".py", encoding="utf-8", delete=False
        ) as tmp:
            tmp_path = Path(tmp.name)
            tmp.write(content)
        yield tmp_path # <~~~ yield == Generator.
    finally:
        os.unlink(tmp_path)
```

If you want me to convert it to `Iterator` I'll do, but I think it shouldn't be changed!

Reply:
`Generator` is a subtype of `Iterator`; https://github.com/python/typeshed/blob/b47213020124bb2e09c143b562205eaf4cde8fe2/stdlib/3/typing.pyi#L131

and `Iterator[Path]` is a better method rather than `Generator[Path, None, None]`, so I introduced this suggestion.

Reply:
Thank you for this information, I've changed it!